### PR TITLE
Implement websocket server!

### PR DESCRIPTION
**Pull request by lezed1**
Add the long rumored websocket server! The WikiSync plugin will attempt to start a websocket server on a port between 37767 and 37776 inclusive. The server will attempt to only accept connections from trusted sources, currently only "localhost", "dps.osrs.wiki", and "tools.runescape.wiki") are allowed.

This PR aims to support everything needed for https://github.com/weirdgloop/osrs-dps-calc/pull/270. The username is broadcasted as a `UsernameChanged` message to all connections as the user logs in/out, and the username is sent when a new connection is opened. The server will also respond to `GetPlayer` requests.

I expect we will want to add support for other origins and message types over time. Hopefully this provides a good foundation for many future WikiSync features!

### CODE REVIEW

**Comment by lezed1** [comment]
We can remove the `// TODO`s once https://github.com/runelite/runelite/pull/17524 is resolved!

File: src/main/java/com/andmcadams/wikisync/dps/DpsDataFetcher.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+		if (!Objects.equals(this.username, currentName))
+		{
+			log.debug("WS player name changed prev=[{}] next=[{}]", this.username, currentName);
+			this.username = currentName;
+			eventBus.post(new UsernameChanged(this.username));
+		}
+	}
+
+	// TODO: Delete this once the Wiki plugin service exists. See https://github.com/runelite/runelite/pull/17524
+	// This is directly copied from https://github.com/runelite/runelite/pull/17524/files#diff-141a15aba5d017de9818b5d39722f85f95b330ef96f8eb06103a947c1094b905
```
Comment (andmcadams) [inline src/main/java/com/andmcadams/wikisync/dps/DpsDataFetcher.java:79]: :+1:

File: src/test/java/com/andmcadams/wikisync/WikiSyncLogPlugin.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		((Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME)).setLevel(Level.WARN);
+		((Logger) LoggerFactory.getLogger("com.andmcadams.wikisync")).setLevel(Level.DEBUG);
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		((Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME)).setLevel(Level.DEBUG);
+	}
+}
```
Comment (andmcadams) [inline src/test/java/com/andmcadams/wikisync/WikiSyncLogPlugin.java:31]: It looks like this is just turning on debug for wikisync only. Is that the case? If so this is a much better way of doing it than we have done in the past lol

**Review by andmcadams — APPROVED** [review]

File: src/test/java/com/andmcadams/wikisync/WikiSyncLogPlugin.java (hunk 2 above)
Comment (lezed1) [reply src/test/java/com/andmcadams/wikisync/WikiSyncLogPlugin.java:31]: Yep! @llemonduck helped me with this

**Pull request merged**